rosbag2_compression/include/rosbag2_compression/zstd_compressor.hpp

```diff
@@ -0,0 +1,48 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef ROSBAG2_COMPRESSION__ZSTD_COMPRESSOR_HPP_
+#define ROSBAG2_COMPRESSION__ZSTD_COMPRESSOR_HPP_
+
+#include <zstd.h>
+
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+#include "base_compressor_interface.hpp"
+#include "visibility_control.hpp"
+
+namespace rosbag2_compression
+{
+
+class ROSBAG2_COMPRESSION_PUBLIC ZstdCompressor : public BaseCompressorInterface
```

Comment:
We had this discussion on the previous PR for the compression interfaces, but I don't see the benefit of splitting compression and decompression in two interfaces. So in this case, on the compression interface is implemented (AFAICT), and I don't understand how users can actually use this as-is.
They are able to compress their bags now which makes them essentially unusable as no means of decompressing it is available. Do I understand this correctly?

Reply:
>I don't see the benefit of splitting compression and decompression in two interfaces.

The reasoning for decoupling was quickly discussed [here](https://github.com/ros2/rosbag2/pull/208#issuecomment-558788634).
> Decoupling allows us to perform specific actions for de/compression which can vary based on the library/algorithm we are using.

For example, we want to measure the performance of playback when we're decompressing without changing the implementation of the compression.

>  I don't understand how users can actually use this as-is.

As is, users can only compress, but the goal is to add decompression as well. We just want to keep PRs small and manageable. We haven't exposed this functionality yet since it's not complete and exposing it via the CLI will be the last step (similar to what was done with splitting).

> They are able to compress their bags now which makes them essentially unusable as no means of decompressing it is available.

There is no means of decompression, but that's being added in the next PR.